### PR TITLE
migrate plugins/slick.cellcopymanager.js to ts

### LINKED ISSUE #10: Migrate slick.cellcopymanager.js

### PR DESCRIPTION
**Pull request by rbusquet**
Closes #10

### CODE REVIEW

File: @scripts/plugins/slick.autotooltips.js [hunk 1]
```diff
@@ -2,32 +2,31 @@
...
-     * @param {boolean} [options.enableForHeaderCells=false] - Enable tooltip for header cells
-     * @param {number}  [options.maxToolTipLength=null]      - The maximum length for a tooltip
+     * @param {boolean} [autoTooltipOptions.enableForCells=true]        - Enable tooltip for grid cells
+     * @param {boolean} [autoTooltipOptions.enableForHeaderCells=false] - Enable tooltip for header cells
+     * @param {number}  [autoTooltipOptions.maxToolTipLength=null]      - The maximum length for a tooltip
      */
-    function AutoTooltips(_a) {
-        var _b = _a.enableForCells, enableForCells = _b === void 0 ? true : _b, _c = _a.enableForHeaderCells, enableForHeaderCells = _c === void 0 ? false : _c, _d = _a.maxToolTipLength, maxToolTipLength = _d === void 0 ? null : _d;
-        var options = { enableForCells: enableForCells, enableForHeaderCells: enableForHeaderCells, maxToolTipLength: maxToolTipLength };
+    function AutoTooltips(autoTooltipOptions) {
```
Comment (rbusquet) [inline @scripts/plugins/slick.autotooltips.js:16]: this was breaking one of the examples. I don't know why cypress was not failing... should we add coverage?

Comment (mitchelvanbever) [reply @scripts/plugins/slick.autotooltips.js:16]: What exactly broke the cypress tests? Just the reference? But might be good to get the CI to run them as well :)

File: plugins/slick.autotooltips.ts [hunk 2]
```diff
@@ -5,55 +5,61 @@ type GenericOption = Record<string, Primitive>;
 // @todo slickGrid type should be in some generic type folder
 type SlickGrid = any;
 
-
```
Comment (mitchelvanbever) [inline plugins/slick.autotooltips.ts:8]: Did you actually make some changes? Hard to tell with all the formatting changes

File: tsconfig.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "module": "commonjs",                           /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
     "lib": ["es6", "dom", "es2017"],                /* Specify library files to be included in the compilation. */
     "allowJs": true,                                /* Allow javascript files to be compiled. */
-    "types": ["node", "jquery", "./lib/jquery.jsonp-2.4.min", "cypress", "@testing-library/cypress"],
+    "types": ["node", "jquery", "jqueryui", "./lib/jquery.jsonp-2.4.min", "cypress", "@testing-library/cypress"],
```
Comment (mitchelvanbever) [inline tsconfig.json:11]: @mpeyper you were right that this would get out of hand fast :p

Comment (mpeyper) [reply tsconfig.json:11]: Yeah, but I'm not sure if there's an alternative way

File: @scripts/plugins/slick.autotooltips.js (hunk 1 above)
Comment (mitchelvanbever) [reply @scripts/plugins/slick.autotooltips.js:16]: nvm, I think see the changes. Thanks for fixing ;)

File: plugins/slick.autotooltips.ts (hunk 2 above)
Comment (mitchelvanbever) [reply plugins/slick.autotooltips.ts:8]: nvm, I think see the changes. Thanks for fixing ;)

**Review by mitchelvanbever — APPROVED** [review]

File: @scripts/plugins/slick.autotooltips.js (hunk 1 above)
Comment (rbusquet) [reply @scripts/plugins/slick.autotooltips.js:16]: Thanks, it was the restructuring in the options breaking when not passing an argument to the constructor.

I'll disable auto formatting in the next PRs

**Pull request merged**